eppo/src/main/java/cloud/eppo/android/ConfigurationRequestor.java

```diff
@@ -7,27 +7,34 @@
 import com.google.gson.JsonIOException;
 import com.google.gson.JsonSyntaxException;
 
-import java.io.Reader;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
 
 import cloud.eppo.android.dto.FlagConfig;
 
 public class ConfigurationRequestor {
     private static final String TAG = logTag(ConfigurationRequestor.class);
 
-    private EppoHttpClient client;
-    private ConfigurationStore configurationStore;
+    private final EppoHttpClient client;
+    private final ConfigurationStore configurationStore;
 
     public ConfigurationRequestor(ConfigurationStore configurationStore, EppoHttpClient client) {
         this.configurationStore = configurationStore;
         this.client = client;
     }
 
     public void load(InitializationCallback callback) {
+        // We have two at-bats to load the configuration: loading from cache and fetching
+        // The below variables help them keep track of each other's progress
+        AtomicBoolean cacheLoadInProgress = new AtomicBoolean(true);
+        AtomicReference<String> fetchErrorMessage = new AtomicReference<>(null);
+        // We only want to fire the callback off once; so track whether or not we have yet
```

Comment:
Pulling back comments and tracking used in the UFC branch as the are more robust than what we have here.